app/models/scheduled_database_check_job.rb

```diff
@@ -0,0 +1,18 @@
+class ScheduledDatabaseCheckJob
+  def self.perform(job_type)
+    case job_type
+    when 'clean_subscription_series'
+      Subscription.where(subscribable_type: "Series").includes(:subscribable).find_each do |s| 
+        s.delete if s.name.nil? && Series.find_by_id(s.subscribable_id).nil?
```

Comment:
2.2.5 :018 > User.where("login IS ?",nil)
  User Load (0.5ms)  SELECT `users`.* FROM `users` WHERE (login IS NULL)
 => [] 
2.2.5 :019 > Series.where("title IS ?",nil)
  Series Load (0.4ms)  SELECT `series`.* FROM `series` WHERE (title IS NULL)
 => [] 
2.2.5 :020 > Work.where("title IS ?",nil)
  Work Load (0.4ms)  SELECT `works`.* FROM `works` WHERE (title IS NULL)
 => [] 

The other check is just be being paranoid when deleting data.